domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/complexfield/ComplexFieldEntityRegularExpressionValidator.java

```diff
@@ -0,0 +1,18 @@
+package uk.gov.hmcts.ccd.definition.store.domain.validation.complexfield;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationResult;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.regex.AbstractRegularExpressionValidator;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.ComplexFieldEntity;
+
+@Component
+public class ComplexFieldEntityRegularExpressionValidator extends AbstractRegularExpressionValidator implements ComplexFieldValidator {
```

Comment:
the code in ComplexFieldEntityRegularExpressionValidator and AbstractRegularExpressionValidator is very specific to Document fields and regex validation on Document fields. So it doesn't look like it's reusable for other complex types and other types of validation on complex types. I would then suggest to remove these 2 classes and merge that code into 1 class `DocumentRegexValidatator` which extends `ComplexFieldValidator`

Reply:
FieldTypeRegularExpressionValidator and ComplexFieldEntityRegularExpressionValidator both extend AbstractRegularExpressionValidator as both need to do that validation. I kept the name generic and not specific to doc type. I can rename all 3 specific to Document Type for e.g. DocTypeRegularExpressionValidator, ComplexFieldDocTypeRegularExpressionValidator and AbstractDocTypeRegularExpressionValidator

Reply:
what I'm saying is that even if you kept the name generic, their code is tied to Documents and Regex validation so it's not generic

Reply:
Renamed